tests/test_interpolation.py

```diff
@@ -11,13 +12,23 @@
     Container,
     DictConfig,
     IntegerNode,
+    ListConfig,
     Node,
     OmegaConf,
     Resolver,
     ValidationError,
+    grammar_parser,
 )
 from omegaconf._utils import _ensure_container
-from omegaconf.errors import ConfigAttributeError, OmegaConfBaseException
+from omegaconf.errors import (
+    ConfigAttributeError,
+    GrammarParseError,
+    OmegaConfBaseException,
+    UnsupportedInterpolationType,
+)
+
+# file deepcode ignore CopyPasteError: there are several tests of the form `c.k == c.k`
+# (this is intended to trigger multiple accesses to the same config key)
```

Comment:
Who is this comment for?
It's not very clear in it's current form. I would remove it and just deal with any deepcode issues with deepcode.

Reply:
`file deepcode ignore CopyPasteError` is the command to disable the DeepCode warning
The comment is supposed to explain it, but maybe it needs to be clearer.
I'll remove that for now and will re-enable if it complains again (it's not a DeepCode issue, it's intended to raise a warning when it sees something like `c.k == c.k`)

Reply:
So here's the Deepcode output: https://www.deepcode.ai/app/gh/omry/omegaconf/d018678a4abbc7ed72ad61c6dcda49879875f781/omry/omegaconf/fd0a5986e039e6120545a5cad4b8ffaf64f159a4/pr/_/%2Ftests%2Ftest_interpolation.py/python%2Fdc%2FCopyPasteError%2Ftest/370/code
(it contains this problem as well as another one I hadn't noticed yet)

Let me know if you prefer me to disable these warnings (with `file deepcode ignore ...`) or just leave them here.

Reply:
we can probably eliminate this by calling __getitem__ there.

```
assert c.__getitem__(k) ==  c.__getitem__(k)
```

Reply:
Hmm that would make code less readable IMO and I doubt it would help, since the DeepCode warning is about "Duplicate expressions on both sides of a binary operator"... and they would still be duplicated.

Reply:
> the DeepCode warning is about "Duplicate expressions on both sides of a binary operator"... and they would still be duplicated.

Just to be sure about that, I gave it a shot, and Deepcode still complains: https://www.deepcode.ai/app/gh/omry/omegaconf/d018678a4abbc7ed72ad61c6dcda49879875f781/omry/omegaconf/51c201f4e45f7842a903b1a1f3fc0f35a792a363/pr/_/%2Ftests%2Ftest_interpolation.py/python%2Fdc%2FCopyPasteError%2Ftest/398/code